### PR TITLE
WINC-1158: Prep for 10.16.0 release

### PR DESCRIPTION
**Pull request by wgahnagl**
This change bumps the operator and the OCP version according to WMCO 10.16.0 release.

Ran:
```
  make bundle
```

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -1,45 +1,21 @@
...
-# backport this content into the old appregistry format and upload it to the quay.io application
-# registry endpoints.
-LABEL com.redhat.delivery.backport=false
-
-# This label gets replaced by render_templates, so it is safer to keep it on its own
-LABEL version="v0.0.0"
-
-# This label maps to the brew build target
-LABEL com.redhat.component="windows-machine-config-operator-bundle-container"
-
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply bundle.Dockerfile:31]: Please undo this change

File: bundle.Dockerfile [hunk 2]
```diff
@@ -1,45 +1,21 @@
...
 LABEL operators.operatorframework.io.bundle.channels.v1=preview,stable
 LABEL operators.operatorframework.io.bundle.channel.default.v1=stable
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.32.0+git
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.32.0
 LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
 LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3
 
+# Labels for testing.
+LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
```
Comment (sebsoto) [inline bundle.Dockerfile:16]: This is fine, as long as it doesn't cause any issues.
We don't make use of scorecard, but if make bundle is generating this I don't see a reason to not have it.

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -169,4 +169,4 @@ spec:
   minKubeVersion: 1.28.0
   provider:
     name: Red Hat
-  version: 0.0.0
+  version: 10.16.0
```
Comment (sebsoto) [inline config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml:172]: Should this change be happening?

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: bundle/metadata/annotations.yaml [hunk 4]
```diff
@@ -9,7 +9,7 @@ annotations:
   operators.operatorframework.io.metrics.builder: operator-sdk-v1.32.0+git
   operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
   operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v3
-  com.redhat.openshift.versions: "=v4.14"
+  com.redhat.openshift.versions: "=v4.16"
```
Comment (mtnbikenc) [inline bundle/metadata/annotations.yaml:12]: Moving this to `v4.16` may be an issue with CVP.  I think that is why master was still at `v4.14`, (which should probably now be updated).  @mansikulkarni96 could you confirm my statement?

Comment (mansikulkarni96) [reply bundle/metadata/annotations.yaml:12]: Correct we can switch to 4.15 now and bump to 4.16 in CY24Q1

File: .gitmodules [hunk 5]
```diff
@@ -15,24 +15,24 @@
 [submodule "kubelet"]
 	path = kubelet
 	url = https://github.com/openshift/kubernetes
-	branch = release-4.15
+	branch = release-4.16
 [submodule "kube-proxy"]
 	path = kube-proxy
 	url = https://github.com/openshift/kubernetes
-	branch = sdn-4.15-kubernetes-1.28.3
+	branch = sdn-4.16-kubernetes-1.28.3
```
Comment (mansikulkarni96) [inline .gitmodules:22]: sdn-4.16-kubernetes-1.28.3 is not available yet, please create a ticket to move to it when it's created.

**Review by mansikulkarni96 — CHANGES_REQUESTED** [review]
Thanks for working on this @wgahnagl

**Review by saifshaikh48 — COMMENTED** [review]
are any of these changes generated by hack scripts? If so I would 
1. break generated changes into their commit
2. include the command that was used in commit message

**Comment by wgahnagl** [comment]
> are any of these changes generated by hack scripts? If so I would
> 
>     1. break generated changes into their commit
> 
>     2. include the command that was used in commit message

Since we changed the versioning system, we'll need to change the hack script to address it, and I figured it would be easier to just do this by hand for now.

File: build/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.15 as build
+FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.16 as build
```
Comment (mansikulkarni96) [inline build/Dockerfile:1]: We are missing these image updates in build/Dockerfile.ci and rest of the other files.

**Comment by mansikulkarni96** [comment]
Please ensure the midstream Dockerfile is updated to use the 4.16 base image.

**Comment by mansikulkarni96** [comment]
I noticed ART has moved to golang 1.21, please verify if the image is available for use and make the necessary updates.

**Review by sebsoto — APPROVED** [review]

**Comment by saifshaikh48** [comment]
@wgahnagl Please create a story to address any required hack script changes so we don't lose track of them

> Since we changed the versioning system, we'll need to change the hack script to address it, and I figured it would be easier to just do this by hand for now.

That's what I meant, split the stuff you updated manually into a separate commit and then manually edit the commit message of the generated commit to include the command you ran to generate it. Was just a suggestion for clarity, not blocking the PR

File: build/Dockerfile [hunk 7]
```diff
@@ -1,4 +1,3 @@
-FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.16 as build
```
Comment (mansikulkarni96) [inline build/Dockerfile]: is this change intended?

**Review by mansikulkarni96 — COMMENTED** [review]
Changes LGTM just one question.
nit: It may be easier for reviews if all the image related updates are in a single commit.

**Comment by wgahnagl** [comment]
> Changes LGTM just one question. nit: It may be easier for reviews if all the image related updates are in a single commit.

I was thinking that it would make sense to break out just the golang bump into its own commit, because if it had to be reverted, it would be harder to do if it were split up in a different way.

File: build/Dockerfile.ci [hunk 8]
```diff
@@ -123,7 +123,7 @@ RUN make build-daemon
 #├── windows_exporter.exe
 #└── windows-instance-config-daemon.exe
 
-FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.15
+FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.16
```
Comment (mansikulkarni96) [inline build/Dockerfile.ci:126]: shouldn't this be 1.21?

**Review by jrvaldes — CHANGES_REQUESTED** [review]
@wgahnagl  Thanks for updating the PR. Mostly LGTM. 

Just missing the version bump in [.ci-operator.yaml](https://github.com/openshift/windows-machine-config-operator/blame/master/.ci-operator.yaml) used in CI

**Review by jrvaldes — APPROVED** [review]
/lgtm
